lib/osf-components/addon/components/file-browser/file-rename-modal/component.ts

```diff
@@ -27,12 +27,12 @@ export default class FileRenameModal extends Component<Args> {
     }
 
     get isValid() {
-        return this.newFileName !== this.originalFileName;
+        return this.newFileName !== this.originalFileName && this.newFileName !== '' && this.newFileName !== null;
```

Comment:
Should this be:

```suggestion
        return this.newFileName !== this.originalFileName && this.newFileName.trim() !== '' && this.newFileName !== null;
```

to catch the whitespace-only names?

Reply:
Also, is there ever a case where it's going to be `null`?

Reply:
This could just check the truthiness of `this.newFileName` first, so something like,
```suggestion
        return Boolean(this.newFileName) && this.newFileName.trim() !== this.originalFileName && this.newFileName.trim() !== '';
```

Reply:
Yes, that would work better if it's possible that newFileName will be null.

Reply:
> Also, is there ever a case where it's going to be `null`? I ask because if it is, it will break the `.trim()`s.

There is an indication that if the value of the input is set to disabled it will be a null value. If this were the case, would invoking trim() within the return statement result in a failure? If so, should we add error handling?

Reply:
Having the first condition be `Boolean(this.newFileName)` should return out of the if statement before it gets to the other conditions if it's `null`, so extra error handling shouldn't be necessary.

Reply:
That sounds good, thank you